### DISCUSSION TITLE
Sujet / Thema 12 –– PDF-Service / Service PDF

### DISCUSSION THREAD

**Discussion by davidoesch** [GeoUnconference/discussions#34] [Ideen / Idées]
**as**                    ProductOwner of a map portal
**I want**             a print service supporting WMS WMTS KML WFS and vectortiles and transform webmercator LV95 to a Preserving area (equal-area or equiareal or equivalent or authalic) Preserving distance (equidistant), projection (LV95),supporting a layout via a YAML style file 

**because**           PDF generation is a need in many portals, operating own mapfish instance can be cumbersome

 

**Acceptance criteria:**

have sth similar as [mapfishprint](https://mapfish.github.io/mapfish-print-doc/) with all functions, callable via a RESTful api/REST Service
user needs to be able to measure distance and bearing on a PDF map
 

**Deliverables:**

feature parity with https://mapfish.github.io/mapfish-print-doc/
scalable
counting generated PDF 
 

**How to demo:**

generate test print via a REST call

**Comment by rastrau** [comment]
During GeoUnconference 3, a group including @maltaesousa from @sitn has analysed this topic and have compared various existing solutions and approaches: 
- mapfish-print
- QGIS Server GetPrint
- report Lab, and
- WeasyPrint

The group's summary can be seen in the following presentation (pdf or markdown): 
[prez.pdf](https://github.com/GeoUnconference/discussions/files/8786756/prez.pdf)
[prez.md](https://github.com/GeoUnconference/discussions/files/8786755/prez.md)

**Comment by ebelo** [comment]
[inkmap ](https://github.com/camptocamp/inkmap), a library for generating high resolution maps in the browser
inkmap is based on [OpenLayers](https://www.openlayers.org/) and will generate maps in PNG format based on a given JSON specification.
inkmap can handle long-running jobs (e.g. A0 format in 300 dpi) and provides an API for following a job progress. It uses a service worker in the background provided the user browser supports [OffscreenCanvas](https://developer.mozilla.org/en-US/docs/Web/API/OffscreenCanvas), and falls back (almost) seamlessly to the main thread if not.

**Comment by eleu** [comment]
mapfishprint is already used with vector tiles today (they are rasterized on the client side and then sent to the print), and, by the way, at Camptocamp we have a generic print server that might be used as SAAS mapfishprint. This said, I rather agree with @ebelo  that the future clearly lies on the side of browser side printing aka. Inkmap ;-)

**Comment by rastrau** [comment]
The group (@procrastinatio, @ltshb and Clemens Rudert) that has worked on this topic during GeoUnconference 4 has [summarised their results – including an outline for a proof-of-concept (PoC) – in a presentation](https://github.com/GeoUnconference/geounconference/blob/main/assets/geounconference-4/topic-12-presentation.pdf): 

[![](https://raw.githubusercontent.com/GeoUnconference/geounconference/main/assets/geounconference-4/topic-12-presentation-title-page.png)](https://github.com/GeoUnconference/geounconference/blob/main/assets/geounconference-4/topic-12-presentation.pdf)
